src/common/Utilities/Containers.h

```diff
@@ -53,17 +53,30 @@ namespace Trinity
             return size;
         }
 
+        // resizes <container> to have at most <requestedSize> elements
+        // if it has more than <requestedSize> elements, the elements to keep are selected randomly
+        // note: after calling this method, the elements remaining in <container> will not necessarily have the same order as they did before the call
         template<class C>
         void RandomResize(C& container, std::size_t requestedSize)
         {
-            uint32 currentSize = uint32(Size(container));
-            while (currentSize > requestedSize)
+            static_assert(std::is_base_of<std::forward_iterator_tag, typename std::iterator_traits<decltype(std::begin(container))>::iterator_category>::value, "Invalid container passed to Trinity::Containers::RandomResize");
```

Comment:
C::iterator is a thing btw, don't nee this ugly decltype (this function obviously cannot be used on arrays)